icons.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
+const childProcess = require('child_process');
+const util = require('util');
+const tempy = require('tempy');
+const gm = require('gm').subClass({imageMagick: true});
+const icns = require('icns-lib');
+
+const filterMap = (map, filterFn) => Object.entries(map).filter(filterFn).reduce((out, [key, item]) => ({...out, [key]: item}), {});
+const baseDiskIconPath = '/System/Library/Extensions/IOStorageFamily.kext/Contents/Resources/Removable.icns';
+
+async function composeIcon(type, appIcon, mountIcon, composedIcon) {
+	mountIcon = gm(mountIcon);
+	appIcon = gm(appIcon);
+	const appIconSize = await util.promisify(appIcon.size).call(appIcon);
```

Comment:
```suggestion
	const appIconSize = await util.promisify(appIcon.size.bind(appIcon)();
```

Same with the other places.